### PR TITLE
Add comprehensive test coverage for init/uninit Claude flows

### PR DESCRIPTION
**Pull request by JMBattista**
## Summary

- **New `utils.test.ts`** (16 tests): First-ever coverage for `copyDirSync`, `addToGitignore`, `removeIfExists`, `removeStaleSmithyArtifacts`
- **New `permissions.test.ts`** (9 tests): Direct tests for `flattenPermissions` covering all entry shapes (`["", "*"]`, nested subcommands, empty args, flag variants)
- **Expanded `claude.test.ts`** (+13 tests): Deploy happy path (all templates written, commands dir, frontmatter stripping, `initPermissions=true/false`), remove accuracy/edge cases (exact count, empty dir, preserve user files, stale commands cleanup), writePermissions idempotency and missing-keys edge case
- **Expanded `cli.test.ts`** (+12 integration tests): Uninit disk verification, settings.json preservation, user file survival, clean-dir handling, full lifecycle (init→uninit→init), double-init idempotency, gitignore creation/dedup/agent-specificity, explicit `--permissions`, issue template content, custom issue template preservation

Test count: **40 → 118**

## Remaining gaps (not automatable)

- **Agent tests (4)**: Require live Claude Code runtime (slash command invocability, permissions enforcement)
- **Human tests (4)**: Interactive Inquirer prompts can't be driven via subprocess

## Test plan

- [x] `npm test` passes — all 118 tests green
- [ ] Verify no regressions in existing CI pipeline

https://claude.ai/code/session_01Ki2Sb2H8k3V4TnuDQY69Nf

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -56,8 +56,19 @@ node dist/cli.js uninit  # Test uninit flow
...
-## Testing Notes
+## Testing
 
-- The CLI uses interactive prompts (Inquirer), so it cannot be tested with piped stdin. Manual testing or a programmatic Node script that simulates the logic is needed.
-- To test slash command behavior in Claude Code: run `smithy init` targeting a test repo with Claude selected, then start a new Claude Code session in that repo and try `/smithy.strike "feature description"`. **Claude Code must be restarted to pick up new/changed commands.**
+Smithy uses a three-tier testing strategy:
+
+1. **Automated** (`npm test`): 122 tests across 7 files covering init/uninit flows, template composition, permissions, and utilities. Runs in CI on every push and PR.
+2. **Agent** (Claude Code session): 4 test cases (A1-A4) verifying prompt visibility, slash command invocability, permissions enforcement, and stale artifact cleanup.
+3. **Human** (interactive terminal): 4 test cases (H1-H4) for Inquirer-based prompts that cannot be driven programmatically.
```
[the remark this answers is not in the document]
Comment (JMBattista) [reply CLAUDE.md]: Good call — removed the hard-coded count in b593c3a. Now says "A comprehensive suite" without specific numbers.

---
_Generated by [Claude Code](https://claude.ai/code)_
